test/SignalR.Orleans.Tests/OrleansHubLifetimeManagerTests.cs

```diff
@@ -118,11 +116,19 @@ public OrleansHubLifetimeManagerTests(OrleansFixture fixture)
         [Fact]
         public async Task InvokeConnectionAsyncOnNonExistentConnectionDoesNotThrow()
         {
+            //TODO: to fix this test requires to mock the `ClientGrain` and set `State.ServerId`

```

Comment:
@galvesribeiro it's not straightforward to fix this tests since we are doing `integration test` here.

This test was failing since `ClientGrain OnConnnect` is not getting called (and this set the `State.ServerId`), so then it was throwing an exception in the `ClientGrain -> SendMessage` validation.

I've added a test below to validate this particular scenario which it will throw an exception.

Reply:
So this will remove the need for the other tests?

I mostly got the tests from Redis backplane implementation. Most of them pass. Just 4 or 5 IIRC fail.

Reply:
there were 5 total tests fails, 2x will be fixed with the other pr since I fixed a bug.
I didn't remove it immediately cause I wanted to double check with you.
I will remove this test and leave the one I've implemented below which basically will be replacing it.

We need to look more in-depth into the two last 2 failing tests which are:
- InvokeAllAsyncWithMultipleServersDoesNotWriteToDisconnectedConnectionsOutput
- InvokeAllAsyncWithMultipleServersWritesToAllConnectionsOutput